core/src/client/mod.rs

```diff
@@ -0,0 +1,58 @@
+use derivative::Derivative;
+
+use crate::{config::Config, metadata::Metadata};
+
+/// Each client should be able to provide access to the following fields
+/// - runtime version
+/// - genesis hash
+/// - metadata
+
+#[derive(Derivative)]
+#[derivative(Debug(bound = ""), Clone(bound = ""))]
+pub struct ClientMetadata<C: Config> {
+    pub genesis_hash: C::Hash,
```

Comment:
Make private? 

I see no reason to make these pub when you have added getters from them